fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/properties/StoredRecordProperty.java

```diff
@@ -217,6 +218,12 @@ public Boolean visitFirstOrDefaultPlan(@Nonnull final RecordQueryFirstOrDefaultP
             return false;
         }
 
+        @Nonnull
+        @Override
+        public Boolean visitDefaultOnEmptyPlan(@Nonnull final RecordQueryDefaultOnEmptyPlan element) {
+            return storedRecordsFromSingleChild(element);
```

Comment:
I don't think we can state a "Maybe" as a result in this visitor, because it is impossible at this point to know whether the underlying child is producing anything or not.
But also faking a stored record if the underlying does not return anything is also misleading IMHO. What do you think?

Reply:
Just say no. :-)